ttcc/core.py

```diff
@@ -4,36 +4,56 @@
 
 def register(device_name, device_details):
     global DEVICES
+
     # Overwrite or ignore a duplicate registration?
     # Check if the device details is in the correct format
     # Check if type is an object of a subclass of TypeBaseClass
-    DEVICES[device_name] = device_details
-
+    
 def parse_device(sentence):
-    pass
+    global DEVICES
+    words = sentence.split()
+    target_device = []
+
+    target_device = list(set(words)&set(DEVICES.keys()))
+    for device in DEVICES:
+            device_alias = list(set(words) & set(DEVICES[device]['alias']))
+            target_device.extend(device_alias)
+    return target_device
+         
+
+def parse_intent(sentence, device):
+    intent = []
+    global DEVICES
```

Comment:
Put all global declarations on the first line of a function, it'll flip out sometimes.